packages/chat-widget/src/index.tsx

```diff
@@ -255,18 +261,55 @@ const MessageGroups: FC<{
         );
       }
 
-      if (response.type === "user" && response.payload.type === "text") {
-        return (
-          <C.MessageGroup key={responseIndex}>
-            <C.Message type="user">
-              <C.MessageBody
-                dangerouslySetInnerHTML={{
-                  __html: marked(response.payload.text),
-                }}
-              />
-            </C.Message>
-          </C.MessageGroup>
-        );
+      if (response.type === "user") {
+        console.log(response);
+        if (response.payload.type === "text") {
+          return (
+            <C.MessageGroup key={responseIndex}>
+              <C.Message type="user">
+                <C.MessageBody
+                  dangerouslySetInnerHTML={{
+                    __html: marked(response.payload.text),
+                  }}
+                />
+              </C.Message>
+            </C.MessageGroup>
+          );
+        }
+
+        if (response.payload.type === "structured") {
+          const { uploadIds, utterance } = response.payload;
+          if (uploadIds == null) {
+            return null;
+          }
+          return (
+            <C.MessageGroup key={responseIndex}>
+              <C.Message type="user">
+                {utterance != null ? (
+                  <C.MessageBody
+                    dangerouslySetInnerHTML={{
+                      __html: marked(utterance),
+                    }}
+                  />
+                ) : null}
+              </C.Message>
+              {uploadIds.map((uploadId) => {
+                const file = props.uploadedFiles[uploadId];
+                if (file != null) {
+                  return (
+                    <img
+                      key={uploadId}
+                      style={{ width: "100%" }}
```

Comment:
```suggestion
                      style={{ maxWidth: "100%" }}
```

in case its a tiny picture.

Also perhaps it should have rounded borders?